core/logx/fields.go

```diff
@@ -1,11 +1,38 @@
 package logx
 
-import "context"
+import (
+	"context"
+	"sync/atomic"
+)
 
-var fieldsContextKey contextKey
+var (
+	fieldsContextKey contextKey
+	// we store *[]LogField as the value, because []LogField is not comparable,
+	// we need to use CompareAndSwap to compare the stored value.
+	globalFields atomic.Value
+)
 
 type contextKey struct{}
 
+// AddGlobalFields adds global fields.
+func AddGlobalFields(fields ...LogField) {
+	for {
+		old := globalFields.Load()
+		if old == nil {
+			val := append([]LogField(nil), fields...)
+			if globalFields.CompareAndSwap(old, &val) {
+				return
+			}
+		} else {
+			oldFields := old.(*[]LogField)
```

Comment:
seems we can simply the code here

Reply:
Use `sync.RWMutex` ?

Reply:
I mean, the if and else code is almost the same, maybe we can combine it.